src/rgw/rgw_http_client.cc

```diff
@@ -600,6 +600,10 @@ int RGWHTTPClient::init_request(rgw_http_req_data *_req_data)
       // TODO: set to size smaller than 1MB should prevent the "Expect" field
       // from being sent. So explicit removal is not needed
       h = curl_slist_append(h, "Expect:");
+      if (!content_type.empty()) {
+        // set content type in case of POST
+        h = curl_slist_append(h, content_type.c_str());
+      }
```

Comment:
why does this need to be specific to `method == "POST"`?

can we just pass in the `Content-Type` as a normal header? keystone auth is doing this with:
```c++
token_req.append_header("Content-Type", "application/json");
```

Reply:
ok. will add as a regular header

Reply:
added regardless of the method is "POST".
not sure what is the difference between the two ways of adding to the header?